packages/eip-1193-provider/src/utils.ts

```diff
@@ -31,16 +31,20 @@ export const preprocessTransaction = ({
     defaultValue: T,
   ): T => (value !== undefined ? converter(value) : defaultValue);
 
-  const typeMapping: { [key: string]: string } = {
-    "0x0": "",
+  const typeMapping: { [key: string]: TransactionType } = {
+    "0x0": "legacy",
     "0x1": "eip2930",
     "0x2": "eip1559",
+    // TODO: enable these once ready
+    // "0x3": "eip4844",
+    // "0x4": "eip7702",
   };
+
   const processedTransaction: TransactionSerializable = {
     ...transaction,
     // @ts-ignore
     chainId: parseInt(transaction.chainId, 16),
-    type: typeMapping[transaction.type ?? ""] ?? "eip1559",
```

Comment:
Any reason we defaulted to eip1559 before and now defaulting to legacy?

Reply:
I was thinking it might be "safer" to default to legacy, but at this point in time it's probably more likely that eip1559 is the most common + likely case. Pushed!